plugins/org.eclipse.emf.ecoretools.ale.ide/src/org/eclipse/emf/ecoretools/ale/ide/project/impl/AleAware.java

```diff
@@ -10,48 +10,80 @@
  *******************************************************************************/
 package org.eclipse.emf.ecoretools.ale.ide.project.impl;
 
-import static org.eclipse.emf.ecoretools.ale.core.preferences.AleProjectPreferences.ALE_SOURCE_FILES;
-import static org.eclipse.emf.ecoretools.ale.core.preferences.AleProjectPreferences.CONFIGURED_FROM_DSL_FILE;
-import static org.eclipse.emf.ecoretools.ale.core.preferences.AleProjectPreferences.DSL_FILE_PATH;
-import static org.eclipse.emf.ecoretools.ale.core.preferences.AleProjectPreferences.ECORE_MODEL_FILES;
+import static java.util.Objects.requireNonNull;
+import static org.eclipse.emf.ecoretools.ale.ide.project.AleProjectPreferences.ALE_SOURCE_FILES;
+import static org.eclipse.emf.ecoretools.ale.ide.project.AleProjectPreferences.CONFIGURED_FROM_DSL_FILE;
+import static org.eclipse.emf.ecoretools.ale.ide.project.AleProjectPreferences.DSL_FILE_PATH;
+import static org.eclipse.emf.ecoretools.ale.ide.project.AleProjectPreferences.ECORE_MODEL_FILES;
 
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Optional;
 
 import org.eclipse.core.resources.IFile;
 import org.eclipse.core.resources.IProject;
 import org.eclipse.core.resources.IResource;
 import org.eclipse.core.resources.ProjectScope;
 import org.eclipse.core.runtime.CoreException;
-import org.eclipse.core.runtime.IPath;
-import org.eclipse.core.runtime.IProgressMonitor;
 import org.eclipse.core.runtime.preferences.IEclipsePreferences;
 import org.eclipse.core.runtime.preferences.IScopeContext;
 import org.eclipse.emf.common.util.URI;
-import org.eclipse.emf.ecoretools.ale.core.interpreter.IAleEnvironment;
-import org.eclipse.emf.ecoretools.ale.core.interpreter.impl.RuntimeAleEnvironment;
-import org.eclipse.emf.ecoretools.ale.core.parser.Dsl;
+import org.eclipse.emf.ecoretools.ale.core.env.IAleEnvironment;
+import org.eclipse.emf.ecoretools.ale.core.env.impl.DslConfiguration;
 import org.eclipse.emf.ecoretools.ale.ide.Activator;
-import org.eclipse.emf.ecoretools.ale.ide.Normalized;
-import org.eclipse.emf.ecoretools.ale.ide.project.AleProject;
+import org.eclipse.emf.ecoretools.ale.ide.project.IAleProject;
 
 /**
- * Decorates {@link IProject} to enhance them with ALE-specific features.
+ * Decorates an {@link IProject} to enhance them with ALE-specific features.
  */
-public class AleAware implements AleProject {
+public class AleAware implements IAleProject {
```

Comment:
I know that this file was modified only due to other renaming but I use the opportunity to suggest
a cleared name for this class: something like _AleAwareProjectDecorator_ or _AleAwareProject_

Reply:
As said in another comment I believe that naming decorators that way makes the code more declarative and readable:
```java
static IAleProject from(IProject project) {
    return new AleAware(project);
}
```
reads way better than:
```java
static IAleProject from(IProject project) {
    return new AleAwareProjectDecorator(project);
}
```
But since this class actually implements `IAleProject` rather than `IProject` I acknowledge that it's not really a decorator and that its name may not fit. That's also the reason why I dislike `AleAwareProject`: an `IAleProject` is obviously "ALE aware".

Since we use the `I` prefix for interfaces I would merely name this class `AleProject`. Does it seem good?

Reply:
Renamed as `AleProject`.